app/controllers/recipes_controller.rb

```diff
@@ -1,42 +1,34 @@
 class RecipesController < ApplicationController
-  before_action :set_recipe, only: [:show]
+  before_action :set_recipe, only: [:show, :edit, :update, :destroy]
   def index
     @recipes = Recipe.all
   end
 
   def new
     @recipe = Recipe.new
-    @recipe.recipe_ingredients.build
+    ri = @recipe.recipe_ingredients.build
+    ri.build_ingredient(params[:recipe_ingredients_attributes])
```

Comment:
```suggestion
    recipe_ingredient = @recipe.recipe_ingredients.build
    recipe_ingredient.build_ingredient(params[:recipe_ingredients_attributes])
```
Avoid 2 letter variable names. It's ok for local development but hard to read for other devs or future you